ddmrp/models/stock_buffer.py

```diff
@@ -1218,6 +1218,7 @@ def cron_actions(self, only_nfp=False):
         """This method is meant to be inherited by other modules in order to
         enhance extensibility."""
         self.ensure_one()
+        self.refresh()
         if not only_nfp or only_nfp == "out":
             self._calc_qualified_demand()
```

Comment:
This method is also used for the nfp calculation, with the refresh in that position it will always be executed, even when you only want to update NFP.